### ISSUE TITLE
Splited datas by 'RandomLinkSplit' have duplicated information.

### ISSUE THREAD

**Issue by Cafelatte1** [pyg-team/pytorch_geometric#6571]
### 🛠 Proposed Refactor

I found a something inefficient point in 'RandomLinkSplit'.
**I think node feature (data.x) and edge information (data.edge_index) are not neccesary for all splited data class.**
Just one data class only have to get the node feature and edge information.

![image](https://user-images.githubusercontent.com/57206771/216317282-8498fe67-04ce-4276-a75d-1f5aa8d8f8f0.png)

### Suggest a potential alternative/fix

How about change the output of function to yield only 'label' and 'splited edge index' like below?
Input on this function already have node feature and entire edge information.

![image](https://user-images.githubusercontent.com/57206771/216319138-16ab2f53-dad5-4c86-81fd-0d6ad6d91d24.png)

**Comment by rusty1s** [comment]
I am happy to include this via an additional flag, such as `keep_attrs=True/False`. WDYT?

**Comment by Cafelatte1** [comment]
> I am happy to include this via an additional flag, such as `keep_attrs=True/False`. WDYT?

'keep_attrs' seems not to be in 'RandomLinkSplit''s arguments.
I am using the 1.12.1+cpu version.

**Comment by rusty1s** [comment]
Yes, I mean we would need to work on including this :)

**Comment by Cafelatte1** [comment]
> Yes, I mean we would need to work on including this :)

Ah ! I misunderstood :)
Yes it is a good simple way which includes an argument like that !

**Comment by rusty1s** [comment]
Do you have interest in contributing this? It isn't super high prio for me, so I would appreciate some help.

**Comment by Cafelatte1** [comment]
> Do you have interest in contributing this? It isn't super high prio for me, so I would appreciate some help.

Yes ! I am interested in contributing for this. Can you guide me about the process which improve this?

**Comment by rusty1s** [comment]
Did you take a look at [`CONTRIBUTING.md`](https://github.com/pyg-team/pytorch_geometric/blob/master/CONTRIBUTING.md)? Happy to answer any remaining questions.
